### PR TITLE
IOS: Return LinkSuccess in onSuccess

### PR DESCRIPTION
**Pull request by amytang0**
Changed the approach of piping from the native layer to the JS layer to be more similar to Android, which is to pipe to two listeners from the objective-C layers’ onSuccess and onExit.
Thus, we indirectly rely on Objective C’s type system

Test plan:
LinkSuccess with user_good, pass_good returns in onSuccess
Success:  
```
{
  "metadata": {
    "accounts": [[Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object]],
    "institution": {
      "id": "ins_6",
      "name": "US Bank"
    },
    "linkSessionId": "e26f9ae7-45d3-400f-a953-67438c5e080a",
    "metadataJson": "{<TRUNCATED>"
  },
  "publicToken": "public-sandbox-499010af-7e82-46ca-8725-ece3e823c7f9"
}
```

LinkExit returns in onExit
  Exit:  
```
  {
  "error": {
    "errorCode": "",
    "errorDisplayMessage": "",
    "errorMessage": "",
    "errorType": ""
  },
  "metadata": {
    "institution": {
      "id": "ins_9",
      "name": "Capital One"
    },
    "linkSessionId": "d1b0f07b-bfd3-4e90-ab6f-1fd80116bbfc",
    "metadataJson": "{\"institution\":{\"name\":\"Capital One\",\"institution_id\":\"ins_9\"},\"request_id\":null,\"link_session_id\":\"d1b0f07b-bfd3-4e90-ab6f-1fd80116bbfc\",\"status\":\"requires_credentials\"}",
    "requestId": "",
    "status": "requires_credentials"
  }
}
```

### CODE REVIEW

**Comment by pladaub** [comment]
This could end up biting us if the SDK's don't send the public token for some reason, but I agree it looks nicer and we're already beholden to the SDKs sending the correct values. LGTM

**Comment by amytang0** [comment]
In the case that we don't send a public_token, that's an error anyway, so it's good that it ends up in LinkExit :)

**Comment by pladaub** [comment]
Yes, but that's an error that could be handled by actually doing validation of the inputs here. Reasonable to take that pass as a follow up

File: ios/RNLinksdk.m [hunk 1]
```diff
@@ -39,7 +39,8 @@
 @interface RNLinksdk ()
 @property (nonatomic, strong) id<PLKHandler> linkHandler;
 @property (nonatomic, strong) UIViewController* presentingViewController;
-@property (nonatomic, strong) RCTResponseSenderBlock completionCallback;
```
Comment (amytang0) [inline ios/RNLinksdk.m:42]: completionCallback gets split up into successCallback and exitCallback, and called in the respective ObjC onSuccess and onExit methods

**Review by afh — APPROVED** [review]
Excellent!

**Pull request merged**